### ISSUE TITLE
log validation loss

### ISSUE THREAD

**Issue by lacava** [lacava/feat#168]
at the moment the log file only captures training loss per generation; update to keep validation loss of best model so far. this will help identify the start of overfitting.
